bundles/org.openhab.automation.pythonscripting/src/main/resources/lib/openhab/helper.py

```diff
@@ -111,7 +127,8 @@ def __call__(self, clazz):
                 _valid_items[cfg.get("groupName") ] = True
             _raw_triggers.append(trigger.raw_trigger)
 
-        clazz.execute = proxy.executeWrapper(clazz.execute, _valid_items)
+        if not isfunction(clazzOrFunc):
+            clazzOrFunc.execute = proxy.executeWrapper(clazzOrFunc.execute, _valid_items)
```

Comment:
If you don't use the wrapper, you lose the following things

1. runtime messuring
2. embedded profiler support @rule(profile=1)
3. logging information about the reason why this rule is triggered
4. additional filtering of unrelated events

Reply:
Intent is to use the wrapper, I was wondering if we just wrap directly in the execute function of the baserule class. Wanted to get your thoughts here.

Reply:
I agree, we could try to simplify something here. I would keep it in a separate function for readability, but instead of swapping we can try to call it in BaseSimpleRule.execute. But lets work on this after the merge of your function based rule changes. Keeping things separately makes it easier to merge :-)